### DISCUSSION TITLE
Custom Task states

### DISCUSSION THREAD

**Discussion by harsh161289** [hibiken/asynq#489] [General]
We want to process streams of input through task queues, so I was looking some help for following items:

1. Depict the progress of a task in someway like metadata which denotes how much data is processed or if stream has stopped sending datasets. Something like: https://docs.celeryq.dev/en/master/userguide/tasks.html#custom-states
2. Delete the queue once a task is complete

**Comment by gschurck** [comment]
Highly interested with this feature to show the progress of the task to the user, is it possible to set a custom state easily currently ?
Or just custom metadata ?
